File: classifier/data/raw/ai/samples/project_euler_java/Problem073.java
// Project Euler 73: count reduced fractions strictly between 1/3 and 1/2.
public class Problem073 {
    public static void main(String[] args) {
        final int limit = 12_000;
        int count = 0;
        for (int denominator = 2; denominator <= limit; denominator++) {
            int first = denominator / 3 + 1;
            int last = (denominator - 1) / 2;
            for (int numerator = first; numerator <= last; numerator++) {
                if (gcd(numerator, denominator) == 1) {
                    count++;
                }
            }
        }
        System.out.println(count);
    }

    private static int gcd(int a, int b) {
        while (b != 0) {
            int t = a % b;
            a = b;
            b = t;
        }
        return a;
    }
}

File: classifier/data/raw/human/samples/project_euler_zig/problem_073.zig
// Project Euler Problem 73: count reduced fractions strictly between 1/3 and 1/2.
const std = @import("std");

pub fn main() !void {
    const limit: u64 = 12_000;
    // The successor of 1/3 in Farey(limit) satisfies 3c - d = 1.
    var a: u64 = 1;
    var b: u64 = 3;
    var c: u64 = (limit + 1) / 3;
    var d: u64 = 3 * c - 1;
    var count: u64 = 0;

    while (2 * c < d) {
        count += 1;
        const k = (limit + b) / d;
        const e = k * c - a;
        const f = k * d - b;
        a = c;
        b = d;
        c = e;
        d = f;
    }

    var buffer: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(std.Io.Threaded.global_single_threaded.io(), &buffer);
    try out.interface.print("{d}\n", .{count});
    try out.interface.flush();
}
